binary/src/Cardano/Binary/Annotated.hs

```diff
@@ -91,3 +102,53 @@ class Decoded t where
 instance Decoded (Annotated b ByteString) where
   type BaseType (Annotated b ByteString) = b
   recoverBytes = annotation
+
+-------------------------------------------------------------------------
+-- Annotated Decoder
+-------------------------------------------------------------------------
+
+
+-- | An AnnotatedDecoder produces a value which needs a reference to the original ByteString to be
```

Comment:
I'm probably slow, but I still struggle to understand exactly what's meant here. Maybe it's possible to include a small example?